crates/bevy_utils/src/anystack.rs

```diff
@@ -0,0 +1,194 @@
+struct AnyStackMeta<T> {
+    offset: usize,
+    func: unsafe fn(value: *mut u8, user_data: &mut T),
+}
+
+pub struct AnyStack<T> {
```

Comment:
If we're going to keep this exported into `bevy_utils`, we should have a doc comment explaining it.